src/snowflake/snowpark/_internal/compiler/utils.py

```diff
@@ -207,20 +210,48 @@ def update_resolvable_node(
             update_resolvable_node(node.source_plan, query_generator)
         resolve_and_update_snowflake_plan(node, query_generator)
 
-    elif isinstance(node, (SelectStatement, SetStatement)):
+    elif isinstance(node, SelectStatement):
+        # clean up the cached sql query and snowflake plan to allow
```

Comment:
do we need to reset that forSnowflakePlan node also?

Reply:
i think we can simply do it before if isinstance(node, SnowflakePlan) for all type of nodes, right?

Reply:
Snowflake plan actually copies it over from the new resolved snowflake plan which already sets it to None.